web/src/components/netflow-record/record-panel.tsx

```diff
@@ -0,0 +1,142 @@
+import './record-panel.css';
+
+export type RecordDrawerProps = {
+  record?: Record;
+  columns: Column[];
+  filters: Filter[];
+  range: number | TimeRange;
+  options: QueryOptions;
+  setFilters: (v: Filter[]) => void;
+  setRange: (r: number | TimeRange) => void;
+  setQueryOptions: (opts: QueryOptions) => void;
+  onClose: () => void;
+  id?: string;
+};
+
+export const RecordPanel: React.FC<RecordDrawerProps> = ({
+  id,
+  record,
+  columns,
+  filters,
+  range,
+  options,
+  setFilters,
+  setRange,
+  setQueryOptions,
+  onClose
+}) => {
+  const { t } = useTranslation('plugin__network-observability-plugin');
+
+  const onClick = React.useCallback(
+    (col: Column, isDelete: boolean) => {
+      if (!record) {
+        return;
+      }
+
+      if (isDelete) {
+        switch (col.id) {
+          case ColumnsId.timestamp:
+            setRange(DEFAULT_TIME_RANGE);
+            break;
+          case ColumnsId.flowdir:
+            setQueryOptions({ ...options, reporter: flowdirToReporter[DEFAULT_FLOWDIR] });
+            break;
+          default:
+            setFilters(filters.filter((f: Filter) => f.colId !== col.id));
+            break;
+        }
+      } else {
+        const value = col.value(record);
+        switch (col.id) {
+          case ColumnsId.timestamp:
+            //Filter at exact same date in ms
+            const dateSeconds = getDateMsInSeconds(Number(value));
+            setRange({ from: dateSeconds, to: dateSeconds + 1 });
+            break;
+          case ColumnsId.flowdir:
+            setQueryOptions({ ...options, reporter: flowdirToReporter[value] });
+            break;
+          default:
+            const values = [{ v: value.toString() }];
+            const result = _.cloneDeep(filters);
+            const found = result.find(f => f.colId === col.id);
+            if (found) {
+              found.values = values;
+            } else {
+              result.push({ colId: col.id, values: values });
+            }
+            setFilters(result);
+            break;
+        }
+      }
+    },
+    [filters, options, record, setFilters, setQueryOptions, setRange]
+  );
+
+  const getFilter = (col: Column) => {
+    let isDelete = false;
+    if (record) {
+      const value = col.value(record);
+      switch (col.id) {
+        case ColumnsId.timestamp:
+          isDelete = typeof range !== 'number' && range.from === getDateMsInSeconds(Number(value));
+          break;
+        case ColumnsId.flowdir:
+          isDelete = options.reporter === flowdirToReporter[value];
+          break;
+        default:
+          isDelete =
+            filters.find((f: Filter) => f.colId === col.id && f.values.find(v => v.v === value.toString())) !==
+            undefined;
+          break;
+      }
+    }
+    return {
+      onClick,
+      isDelete
+    };
+  };
+
+  return (
+    <DrawerPanelContent id={id}>
+      <DrawerHead>
+        <Text component={TextVariants.h2}>{t('Traffic details')}</Text>
```

Comment:
maybe "Flow Details" rather than "Traffic details"? 
Also mind the capital "Details" , as per the "title case" guidelines here: https://www.patternfly.org/v4/ux-writing/capitalization/#capitalization-types